storage/mysql/map_storage.go

```diff
@@ -124,21 +124,22 @@ func (m *mapTX) Set(keyHash []byte, value trillian.MapLeaf) error {
 	}
 	defer stmt.Close()
 
-	_, err = stmt.Exec(m.ms.mapID, []byte(keyHash), m.writeRevision, flatValue)
+	_, err = stmt.Exec(m.ms.mapID, keyHash, m.writeRevision, flatValue)
 	return err
 }
 
-func (m *mapTX) Get(revision int64, keyHashes [][]byte) ([]trillian.MapLeaf, error) {
-	stmt, err := m.ms.getStmt(selectMapLeafSQL, len(keyHashes), "?", "?")
+// TODO: API problem: there's nothing to say which map leaves correspond to which indexes.
+func (m *mapTX) Get(revision int64, indexes [][]byte) ([]trillian.MapLeaf, error) {
+	stmt, err := m.ms.getStmt(selectMapLeafSQL, len(indexes), "?", "?")
 	if err != nil {
 		return nil, err
 	}
 	stx := m.tx.Stmt(stmt)
 	defer stx.Close()
 
-	args := make([]interface{}, 0, len(keyHashes)+2)
-	for _, k := range keyHashes {
-		args = append(args, []byte(k[:]))
+	args := make([]interface{}, len(indexes), len(indexes)+2)
+	for i, index := range indexes {
```

Comment:
Under the hood, I think they're doing something extremely similar, which explains why their performance is similar. For me this is more about code readability.